app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt

```diff
@@ -2,25 +2,80 @@ package org.dicekeys.app.viewmodels
 
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import dagger.hilt.android.lifecycle.HiltViewModel
+import androidx.lifecycle.ViewModelProvider
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import org.dicekeys.api.ApiStrings
+import org.dicekeys.app.repositories.DiceKeyRepository
 import org.dicekeys.dicekey.DiceKey
 import org.dicekeys.dicekey.Face
-import javax.inject.Inject
-
-@HiltViewModel
-class ApiRequestViewModel @Inject constructor() : ViewModel() {
 
+class ApiRequestViewModel @AssistedInject constructor(
+    private val diceKeyRepository: DiceKeyRepository,
+    @Assisted val command: String
+) : ViewModel() {
     var diceKey = MutableLiveData<DiceKey<Face>>()
 
+    val sequence = MutableLiveData("")
+
     var title = MutableLiveData("")
+    var subtitle = MutableLiveData("")
     var approve = MutableLiveData("")
 
     var recipe = MutableLiveData("")
 
     var createLabel = MutableLiveData("")
     var dataCreated = MutableLiveData("")
 
+    val showSequence = MutableLiveData(true)
+
+    val hideFaces = diceKeyRepository.hideFaces
+
+    init {
+        showSequence.value = when (command) {
+            ApiStrings.Commands.getPassword, ApiStrings.Commands.getSecret -> {
+                true
+            }
+            else -> false
+        }
+    }
+
+    fun toggleHideFaces() {
+        diceKeyRepository.toggleHideFaces()
+    }
+
+    fun sequenceUp() {
+        updateSequence((sequence.value?.toIntOrNull() ?: 1) + 1)
+    }
+
+    fun sequenceDown() {
+        updateSequence((sequence.value?.toIntOrNull() ?: 2) - 1)
```

Comment:
will this change sequence to null if going down from 2?

Reply:
Fixed in ApiRequestViewModel & RecipeBuilder to be consistent.